### PR TITLE
Prevent backspace from deleting entities

### LINKED ISSUE #69: Prevent backspace from deleting entities
As a user, I do not want to delete entities when editing text.
As a developer, I do not want to delete entities when editing text.

### PR DESCRIPTION
**Pull request by veikkahonkanen**
Resolves #69

### CODE REVIEW

File: src/components/screens/editor/editor-screen.tsx [hunk 1]
```diff
@@ -53,7 +53,8 @@ const EditorScreen: React.FC<Props> = ({
   const [ leftToolBarIndex, setLeftToolBarIndex ] = React.useState(0);
   const [ rightToolBarIndex, setRightToolBarIndex ] = React.useState(0);
   const [ addingKnots, setAddingKnots ] = React.useState(false);
-  const [ dataChanged, setDataChanged ] = React.useState(false);
+  const [ dataChanged, /* setDataChanged */ ] = React.useState(false);
```
Comment (lauriLukkarinen) [inline src/components/screens/editor/editor-screen.tsx:63]: Why this is commented out?

File: src/components/views/story-editor-view/story-editor-view.tsx [hunk 2]
```diff
@@ -90,6 +92,10 @@ const StoryEditorView: React.FC<Props> = ({
     }
   }, [ addingKnots, newPoint, onAddNode ]);
 
+  React.useEffect(() => {
+    editingEntityInfo ? engineRef.current.getModel().setLocked(true) : engineRef.current.getModel().setLocked(false);
+  }, [editingEntityInfo]);
```
Comment (lauriLukkarinen) [inline src/components/views/story-editor-view/story-editor-view.tsx]: Remember, don't think too complicated, dynamically change only the part that needs to be changed.
```suggestion
  React.useEffect(() => {
    engineRef.current.getModel().setLocked(editingEntityInfo);
  }, [ editingEntityInfo ]);
```

**Review by lauriLukkarinen — CHANGES_REQUESTED** [review]

File: src/components/screens/editor/editor-screen.tsx (hunk 1 above)
Comment (veikkahonkanen) [reply src/components/screens/editor/editor-screen.tsx:63]: To suppress CloudFlare error for unused constant. The setDataChanged constant will be needed later on when implementing the save button functionality.

Comment (lauriLukkarinen) [reply src/components/screens/editor/editor-screen.tsx:63]: 👍🏻

**Review by lauriLukkarinen — APPROVED** [review]

**Pull request merged**